Test data

HadrienG opened the pull request:
- [x] download test genomes
- [x] simulate datasets
- [x] assembly
- [x] mapping
- [x] binning

DrAntoine committed:
edit readme.md

DrAntoine committed:
Create makefile
to download sequences and sorting entry

DrAntoine committed:
Preparation data scripts : done

HadrienG commented on .vscode/settings.json:
```diff
@@ -0,0 +1,6 @@
+{
```
don't commit the config please

HadrienG commented on .gitignore:
```diff
@@ -102,3 +102,6 @@ venv.bak/
 .mypy_cache/
+
+# samples
+samples/*
```
there should be a newline at end of the file

DrAntoine replied:
fixed

HadrienG commented on Makefile:
```diff
@@ -0,0 +1,60 @@
+	@mv samples/*.fna samples/complete_genomes
+	@echo It looks clean :D
+
+chroplasmitor:
```
I hate fun, so please use explicit names ;)

DrAntoine replied:
fixed

HadrienG commented on Pipfile.lock:
```diff
@@ -0,0 +1,173 @@
+{
```
This is not a standalone application so don't commit the lock

DrAntoine replied:
Fixed

HadrienG commented on scripts/chroplasmitor.py:
```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/python3
```
/usr/bin/env python3 is better

DrAntoine replied:
fixed

HadrienG commented on scripts/chroplasmitor.py:
```diff
@@ -0,0 +1,81 @@
+from Bio import SeqIO
+
+
+def writter(path, records):
```
writer, only one t

DrAntoine replied:
fixed

HadrienG commented on scripts/chroplasmitor.py:
```diff
@@ -0,0 +1,81 @@
+def writter(path, records):
+    with open(path, "w") as file:
+        for record in records:
+            file.write(f">{record.description}\n")
```
could have used SeqIO.write here

DrAntoine replied:
fixed but need to be tested

DrAntoine replied:
its working

Commit:
update gitignore

.vscode/ folder untracked and .Pipfile.lock too.
add an empty line at the end of file.

Commit:
remove vscode settings file

Commit:
remove Pipfile.lock

Commit:
add explicite name for rule in Makefile

Commit:
update file.

Add usr/bin/env python3
use SeqIO.write, need to be tested !
fix writter -> writer
Add docstring

DrAntoine commented:
metabat installed
problem during installation of concoct (probably an error between py2 and py3)

Commit:
change the echo sentence in sort_entry rule

HadrienG commented:
they claim that python3 is supported though. Try to use a [conda](https://bioconda.github.io/) environment?

DrAntoine committed:
first implementation of tetranucl clustering,
take a lot of memory,
clustering not yet implemented, just a graph to visualize cluster

DrAntoine committed:
concoct implemented,
metrics in progress
